src/code/z_demo.c

```diff
@@ -1,99 +1,1649 @@
+        case 1:
+            if (csCtx->frames == cmd->startFrame) {
+                func_8013ECE0(0.0f, cmd->unk6, cmd->unk7, cmd->unk8);
+            }
+            break;
+
+        case 2:
+            if ((csCtx->frames >= cmd->startFrame) && (cmd->endFrame >= csCtx->frames) &&
+                ((csCtx->frames == cmd->startFrame) || (globalCtx->state.frames % 64) == 0)) {
+                func_8013ECE0(0.0f, cmd->unk6, cmd->unk7, cmd->unk8);
+            }
+            break;
+    }
+}
+
+// Command 0x9B:
+void Cutscene_Command_FadeColorScreen(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdFadeScreen* cmd) {
+    if ((csCtx->frames >= cmd->startFrame) && (cmd->endFrame >= csCtx->frames)) {
+        f32 alpha;
+
+        globalCtx->envCtx.fillScreen = true;
+        alpha = Environment_LerpWeight(cmd->endFrame, cmd->startFrame, csCtx->frames);
+
+        if (((cmd->unk0 == 1)) || (cmd->unk0 == 2)) {
+            globalCtx->envCtx.screenFillColor[0] = cmd->r;
+            globalCtx->envCtx.screenFillColor[1] = cmd->g;
+            globalCtx->envCtx.screenFillColor[2] = cmd->b;
+
+            if (cmd->unk0 == 2) {
+                globalCtx->envCtx.screenFillColor[3] = (1.0f - alpha) * 255.0f;
+            } else {
+                globalCtx->envCtx.screenFillColor[3] = 255.0f * alpha;
+            }
+        }
+    }
+}
+
+// Command 0x9D: Set Time of Day & Environment Time
+void Cutscene_Command_SetTime(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdDayTime* cmd) {
+    u16 nextTime;
+    u16 hourAsMinutes;
+    u16 minutes;
+
+    if (csCtx->frames == cmd->startFrame) {
+        hourAsMinutes = (cmd->hour * 60.0f) / (60.0f * 24 / 0x10000);
+        minutes = ((cmd->minute + 1) / (60.0f * 24 / 0x10000));
+
+        nextTime = hourAsMinutes + minutes;
+        gSaveContext.time = nextTime;
+        gSaveContext.environmentTime = nextTime;
+    }
+}
+
+void Cutscene_TerminatorImpl(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdBase* cmd) {
+    csCtx->state = CS_STATE_4;
+    func_80165690();
+    Audio_SetCutsceneFlag(false);
+    gSaveContext.unk_3F48 = 1;
+
+    if ((gSaveContext.gameMode != 0) && (csCtx->frames != cmd->startFrame)) {
+        gSaveContext.unk_3F1E = 1;
+    }
+
+    gSaveContext.cutscene = 0;
+    if (cmd->base == 1) {
+        globalCtx->nextEntranceIndex = globalCtx->csCtx.sceneCsList[globalCtx->csCtx.currentCsIndex].nextEntranceIndex;
+        gSaveContext.nextCutsceneIndex = 0;
+        globalCtx->sceneLoadFlag = 0x14;
+        if (gSaveContext.gameMode != 1) {
+            Scene_SetExitFade(globalCtx);
+        } else {
+            D_801BB12C++;
+            if (D_801BB12C >= 2) {
+                D_801BB12C = 0;
+            }
+            globalCtx->unk_1887F = 4;
+        }
+
+        if ((globalCtx->nextEntranceIndex & 0xF) > 0) {
+            gSaveContext.nextCutsceneIndex = (globalCtx->nextEntranceIndex & 0xF) + 0xFFEF;
+        }
+
+        globalCtx->nextEntranceIndex &= 0xFFF0;
```

Comment:
```suggestion
        globalCtx->nextEntranceIndex &= ~0xF;
```
I think this makes sense with above conditional, but could be wrong.